src/algebra/scalar/authenticated_scalar.rs

```diff
@@ -718,16 +732,17 @@ impl<C: CurveGroup> Sub<&Scalar<C>> for &AuthenticatedScalarResult<C> {
     fn sub(self, rhs: &Scalar<C>) -> Self::Output {
         // Party 1 subtracts a zero value from their share to allocate a new ID for the
         // result and stay in sync with party 0
-        let new_share = &self.share - rhs;
```

Comment:
Some of the implementations _appear_ different, they are not. They were previously delegating their implementation to `MpcScalar`, now we inline them for gate-efficiency.